File: .skills/openclaw-skills/skills/githubxiaohei/memory-system-optimizer/scripts/billing.js
#!/usr/bin/env node
/**
 * SkillPay 收费模块
 * 
 * 使用方法：
 *   node billing.js charge <user_id>
 *   node billing.js balance <user_id>
 *   node billing.js payment-link <user_id> <amount>
 */

const SKILLPAY_API = "https://skillpay.me/api/v1/billing";

// 从环境变量读取配置
const API_KEY = process.env.SKILLPAY_API_KEY;
const SKILL_ID = process.env.SKILLPAY_SKILL_ID;

if (!API_KEY || !SKILL_ID) {
  console.error("❌ 请配置环境变量:");
  console.error("   export SKILLPAY_API_KEY=\"你的APIKey\"");
  console.error("   export SKILLPAY_SKILL_ID=\"你的SkillID\"");
  process.exit(1);
}

const command = process.argv[2];
const userId = process.argv[3] || `user_${Date.now()}`;
const amount = parseFloat(process.argv[4]) || 5.0;

async function charge(userId) {
  console.log(`💳 正在扣费... user: ${userId}`);
  
  const response = await fetch(`${SKILLPAY_API}/charge`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      "X-API-Key": API_KEY
    },
    body: JSON.stringify({
      user_id: userId,
      skill_id: SKILL_ID
    })
  });
  
  const result = await response.json();
  
  if (result.success) {
    console.log(`✅ 扣费成功！余额: ${result.balance} USDT`);
    return true;
  } else {
    console.log(`❌ 余额不足！`);
    console.log(`💳 支付链接: ${result.payment_url}`);
    return false;
  }
}

async function getBalance(userId) {
  console.log(`📊 查询余额... user: ${userId}`);
  
  const response = await fetch(`${SKILLPAY_API}/balance?user_id=${userId}`, {
    headers: {
      "X-API-Key": API_KEY
    }
  });
  
  const result = await response.json();
  console.log(`💰 余额: ${result.balance} USDT`);
}

async function createPaymentLink(userId, amount) {
  console.log(`🔗 生成支付链接... user: ${userId}, 金额: ${amount} USDT`);
  
  const response = await fetch(`${SKILLPAY_API}/payment-link`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      "X-API-Key": API_KEY
    },
    body: JSON.stringify({
      user_id: userId,
      amount: amount
    })
  });
  
  const result = await response.json();
  console.log(`💳 支付链接: ${result.payment_url}`);
}

async function main() {
  const command = process.argv[2];
  
  switch (command) {
    case "charge":
      await charge(process.argv[3] || `user_${Date.now()}`);
      break;
    case "balance":
      await getBalance(process.argv[3] || `user_${Date.now()}`);
      break;
    case "payment-link":
      await createPaymentLink(
        process.argv[3] || `user_${Date.now()}`,
        parseFloat(process.argv[4]) || 5.0
      );
      break;
    default:
      console.log(`
Usage: node billing.js <command> [args]

Commands:
  charge <user_id>           扣费（检查余额并扣费）
  balance <user_id>          查询余额
  payment-link <user_id> <amount>  生成充值链接

Examples:
  node billing.js charge user_123
  node billing.js balance user_123
  node billing.js payment-link user_123 10
`);
  }
}

main().catch(console.error);

File: .skills/openclaw-skills/skills/pikachu022700/quant-orchestrator-free/billing.py
#!/usr/bin/env python3
"""
SkillPay Billing SDK - Python
1 USDT = 1000 tokens | 1 call = 1 token | Min deposit 8 USDT
"""

import os
import requests
import json

BILLING_URL = "https://skillpay.me/api/v1/billing"
API_KEY = "sk_72df482063c0d454184ac7677a9f1094638da3e9f9b51530c9ffab92d0427011"
SKILL_ID = "quant-orchestrator"

HEADERS = {
    "X-API-Key": API_KEY,
    "Content-Type": "application/json"
}

def charge_user(user_id: str) -> dict:
    """Charge user for skill usage"""
    try:
        response = requests.post(
            f"{BILLING_URL}/charge",
            json={
                "user_id": user_id,
                "skill_id": SKILL_ID,
                "amount": 1  # 1 token
            },
            headers=HEADERS,
            timeout=10
        )
        data = response.json()
        
        if data.get("success"):
            return {"ok": True, "balance": data.get("balance")}
        else:
            return {
                "ok": False, 
                "balance": data.get("balance"),
                "payment_url": data.get("payment_url")
            }
    except Exception as e:
        return {"ok": False, "error": str(e)}

def get_balance(user_id: str) -> float:
    """Get user balance"""
    try:
        response = requests.get(
            f"{BILLING_URL}/balance",
            params={"user_id": user_id},
            headers=HEADERS,
            timeout=10
        )
        return response.json().get("balance", 0)
    except:
        return 0

def get_payment_link(user_id: str, amount: float = 10) -> str:
    """Get payment link for user"""
    try:
        response = requests.post(
            f"{BILLING_URL}/payment-link",
            json={
                "user_id": user_id,
                "amount": amount
            },
            headers=HEADERS,
            timeout=10
        )
        return response.json().get("payment_url", "")
    except:
        return ""

def check_and_charge(user_id: str) -> dict:
    """Check balance and charge if sufficient"""
    charge_result = charge_user(user_id)
    
    if charge_result.get("ok"):
        return {
            "success": True,
            "message": "Skill executed successfully",
            "balance": charge_result["balance"]
        }
    else:
        payment_url = charge_result.get("payment_url", get_payment_link(user_id))
        return {
            "success": False,
            "message": "Insufficient balance. Please deposit.",
            "payment_url": payment_url,
            "balance": charge_result.get("balance", 0)
        }

# CLI test
if __name__ == "__main__":
    import sys
    if len(sys.argv) > 1:
        user_id = sys.argv[1]
    else:
        user_id = "test_user"
    
    print(f"Testing billing for user: {user_id}")
    result = check_and_charge(user_id)
    print(json.dumps(result, indent=2))
